addons/sale_project/models/sale_order.py

```diff
@@ -203,6 +204,18 @@ def action_view_milestone(self):
             }
         }
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        project = self.env['project.project'].browse(self.env.context.get('create_for_project'))
+        created_records = super().create(vals_list)
+        if project:
+            service_sol = next((sol for sol in created_records.order_line if sol.is_service), False)
+            if not service_sol:
+                raise UserError(_('This Sales Order must contain at least one product of type "Service".'))
+            if not project.sale_line_id:
+                project.sale_line_id = service_sol
+        return created_records
```

Comment:
There is a `default_project_id` on the stat button, so the created orders will be linked to the project in the context

Reply:
What I am mostly referring to is the use of next and the fact that it only returns one element (the next/first one) from the iterator passed to it.
Lets say for example I create a new SO with 2 SOLs, 5 hours of Junior Architect and 2 hours of Senior Architect services - in that order. The service_sol will be the 5 hours of Junior Architect services and yeah it will be correctly linked to the project from the "create_for_project" context variable. But now the second SOL will be kind of gone, the stat button in project.update will only show 1 SOL while its actually 2.